pyproject.toml

```diff
@@ -21,6 +21,10 @@ python = "^3.7"
 # Pin this to the lowest supported version of Nautobot
 nautobot = "~1.2.1"
 
+# Nautobot dependency on pyopenssl requires cryptography<39 https://github.com/pyca/cryptography/issues/7959
+# After updating Nautobot pyopenssl dependency to ">=22.1", remove following pin
```

Comment:
- Current versions of Nautobot don't, as far as I can tell, depend on `pyopenssl` at all.
- Nautobot 1.4.0 had pyopenssl 22.0 in its `poetry.lock` but I don't see anything there that would block 22.1. 
- Same for Nautobot 1.3.0
- I don't see it in the poetry.lock for Nautobot 1.2.1 at all.

What version of Nautobot were you hitting this issue with? We shouldn't pin to an older version of cryptography since there are known CVEs present; if we have to drop support for older Nautobot versions to support newer cryptography, that would be the preferred fix.

Reply:
Alternately could we pin to a newer version of `pyopenssl` rather than an older version of `cryptography`?

Reply:
I'm confused though, why are you having to pin this back at all? Per @snaselj's feedback in Slack we have `cryptography<39` in our lockfile. This seems redundant.

Reply:
I'm a bit baffled by the behavior showing in 1.2.1 in CI - I just spun up the `ghcr.io/nautobot/nautobot-dev:1.2.1-py3.8` Docker image locally and it *does not have* pyopenssl installed at all. Where are we getting it from in CI?

Reply:
The issue is with the latest version of Nautobot v1.5.12.

https://github.com/pyca/cryptography/issues/7959#issuecomment-1368711852

https://github.com/nautobot/nautobot/blob/develop/poetry.lock#L592
https://github.com/nautobot/nautobot/blob/develop/poetry.lock#L2940

Breaking change was introduced in the following PR
https://github.com/nautobot/nautobot/pull/3258

We should fix it in Nautobot, either by upgrading pyopenssl or downgrading cryptography.

Reply:
Oh, it's not the actual pyopenssl, it's the vendored version included in urllib3:

`from urllib3.contrib.pyopenssl import orig_util_SSLContext as SSLContext`

Seeing if I can track down more details.

Reply:
Failing test against `develop` can be found here:
https://github.com/nautobot/nautobot-plugin-secrets-providers/actions/runs/4365503065/jobs/7634210599#step:7:1632

Reply:
I wouldn't propose we re-introduce this vulnerability: https://github.com/advisories/GHSA-x4qr-2fvf-3mr5.

How are we not getting reports of broken installations?

Reply:
This is so weird. The stack trace shows `/usr/local/lib/python3.7/site-packages/OpenSSL/` but that package (actually `pyopenssl` **isn't there in the base Docker image** and I don't see `pyopenssl` being installed anywhere in the CI log... so where is it coming from?

Reply:
I don't think we have code in core that's using either `cryptography` or `pyopenssl` directly, and certainly `botocore` isn't included in the core image. I can confirm that updating `pyopenssl` to `23.0.0` appears to fix this issue.

Reply:
I updated cryptography and pyopenssl to be compatible.